### ISSUE TITLE
onColumnResized weird behavior

### ISSUE THREAD

**Issue by zhenya2224** [ag-grid/ag-grid#1945]
**I'm submitting a ...**  (check one with "x")
```
[x] bug report => search github for a similar issue or PR before submitting
[ ] feature request
[ ] support request => Please do not submit support request here, instead see https://github.com/ceolter/ag-grid/blob/master/CONTRIBUTING.md#question
```

**Current behavior**

I have a function on column resize event which fires only if event.finished:true. Also on gridReady I run function api.sizeColumnsToFit(). 

**Expected behavior**

In version 10.1.0 works fine and  api.sizeColumnsToFit() doesn't trigger event.finished:true onColumnResized event, but since this version(I've tried ag-grid 11,12,13) the behavior has changed on grid  initialisation I trigger   api.sizeColumnsToFit() and it forces the grid onColumnResized event add finished:true

**Minimal reproduction of the problem with instructions**
In version 10.1.0 the behavior is correct and the event object if I log
onColumnResized: function(event) {
                                console.log(event)
} Looks like this
{
e {type: "columnResized", column: e}
column:e {moving: false, filterActive: false, eventService: e, rowGroupActive: false, pivotActive: false, …}
type:"columnResized"
}  - if I dont try to resize column it doesnt have finished:true property

But when I switch to version 12 for example, the event object if I log
onColumnResized:  function(event) {
 console.log(event)
} Looks like this
{
type: "columnResized",
 column: e, columns: Array(1),
 finished: true, api: e, …
}

It has finished true property even if I haven't tried to resize the column. If api.sizeColumnsToFit() is called it forces the whole grid into the infinite loop.

**Plunkr**
https://plnkr.co/edit/7PwbmB6HYAOyPopXqpNL?p=preview (check console.log) version 13
https://plnkr.co/edit/iH7coI7fGDIS7Kbrp98w?p=preview version 10.1

**What is the motivation / use case for changing the behavior?**

It forces the whole grid in infinitive loop.
**Please tell us about your environment:**

* **ag-Grid version:** 13.0.0

* **Browser:**"Mozilla/5.0 (Windows NT 6.3; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/63.0.3218.0 Safari/537.36"
* **Language:** [all | TypeScript X.X | ES6/7 | ES5]
Angular 1.6

**Comment by rmc-software** [comment]
Hi,

There hasn't been any functional change other than adding extra info to the event.

I don't see the infinite loop you're referring to either?

The reason why finished:true is because the resize has completed when onColumnResized() is called. Note that if you drag a column it will be false until you let go of the mouse.

**Comment by zhenya2224** [comment]
In version 10 finished true wasn't called you can check codepen console log

**Comment by rmc-software** [comment]
We have reviewed this functionality and are satisfied we don't have any defects around this area.

If you're an Enterprise customer you can use our Members Forum where we can provide more guidance. https://ag-grid.com/forum/forumdisplay.php?fid=5

Thanks

**Comment by zhenya2224** [comment]
Ok thanks for your reply, so its normal behavior that column hasn't been dragged but fires with (finished: true) with onColumnResized()?

**Comment by rmc-software** [comment]
correct.

**Comment by zhenya2224** [comment]
Ok, thanks again, last question, but is possible to distinguish if the column resize event was triggered by  this.api.sizeColumnsToFit() or by physically dragging with mouse?

**Comment by rmc-software** [comment]
As per the issue template, please don't post support queries here. Use either stackoverflow.com, or if you're an Enterprise customer you can also use our Members Forum here: https://ag-grid.com/forum/forumdisplay.php?fid=5

**Comment by zhenya2224** [comment]
Already tried, here the the man with same question
https://stackoverflow.com/questions/45334312/ag-grid-how-to-tell-if-column-resize-is-done-manually-vs-automatically-with-on
I'll think about enterprise

**Comment by twbutler** [comment]
FWIW, I am having the exact same problem reported by @zhenya2224 with version 16.  Previously we were using version 9, and I had implemented onColumnResized like so:

    onColumnResized: (params: any) => {
      // if a column has been resized by the user, resize all columns to fit
      if (params.type === 'columnResized' && params.finished === true) {
         this.gridOptions.api.sizeColumnsToFit();
      }
    },

This was very useful in keeping the table looking tidy (no horizontal scrollbar).  However, now with newer ag-grid versions, the call to sizeColumnsToFit() causes some kind of infinite loop and the browser comes to a crawl.  This behavior is very reproducible.  However, since it seems that the problem will not be fixed, I have had to remove my onColumnResized() to get normal behavior again, but if the user resizes a column, the horizontal scrollbar appears (since sizeColumnsToFit() is not being called).
